FilePersistence/src/version_control/Commit.cpp

```diff
@@ -82,8 +82,13 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 	else
 	{
 		iter = files_.constBegin();
-		while (!iter.key().contains(fileName) && iter != files_.constEnd())
+		while (iter != files_.constEnd())
```

Comment:
This check is still somewhat brittle, as it still relies on specifics of the name, which might change in the future.

Why not use `QFileInfo::isDir` for entries that contain `fileName`? This should be more stable.